packages/@react-aria/landmark/src/useLandmark.ts

```diff
@@ -0,0 +1,182 @@
+      
+      // If something was previously focused in the next landmark, then return focus to it
+      if (nextLandmark.lastFocused) {
+        let lastFocused = nextLandmark.lastFocused;
+        if (document.body.contains(lastFocused)) {
+          lastFocused.focus();
+          return;
+        }
+      }
+
+      // Otherwise, focus the first focusable element in the next landmark
+      let walker = getFocusableTreeWalker(nextLandmark.ref.current, {tabbable: true});
+      let nextNode = walker.nextNode() as HTMLElement;
+      if (!nextNode) {
+        nextNode = walker.firstChild() as HTMLElement;
+      }
+      nextNode.focus();
+    }
+  }
+
+  // Only the closest landmark cares about focus inside of it
+  public focusinHandler(e: FocusEvent) {
+    let currentLandmark = this.closestLandmark(e.target as HTMLElement);
+    this.landmarks = this.landmarks.map((landmark) => ({...landmark, ...(landmark === currentLandmark && {lastFocused: e.target as HTMLElement})}));
+  }
+}
+
+document.addEventListener('keydown', LandmarkManager.getInstance().f6Handler.bind(LandmarkManager.getInstance()), {capture: true});
+document.addEventListener('focusin', LandmarkManager.getInstance().focusinHandler.bind(LandmarkManager.getInstance()), {capture: true});
+
+/**
+ * Provides landmark navigation in an application. Call this with a role and label to register a landmark navigable with F6.
+ * @param props - Props for the landmark.
+ * @param ref - Ref to the landmark.
+ */
+export function useLandmark(props: AriaLandmarkProps, ref: MutableRefObject<HTMLElement>): LandmarkAria {
+  const {
+    role,
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledby
+  } = props;
+  let manager = LandmarkManager.getInstance();
+  let label = ariaLabel || ariaLabelledby;
+
+  useLayoutEffect(() => {
+    manager.addLandmark(ref, {label, role});
+
+    // Warn if there are 2+ landmarks with the same role but no label.
+    // Labels for landmarks with the same role must also be unique.
+    // https://www.w3.org/TR/wai-aria-practices/examples/landmarks/navigation.html
+    // TODO: Handle duplicate labels more precisely.
+    let landmarksWithRole = manager.getLandmarksByRole(role);
+    if (landmarksWithRole.size > 1) {
+      if ([...landmarksWithRole].some(landmark => !landmark.label)) {
+        console.warn(`Page contains more than one landmark with the '${role}' role. If two or more landmarks on a page share the same role, all must be labeled with an aria-label or aria-labelledby attribute.`);
+      } else if ([...landmarksWithRole].filter(landmark => landmark.label === label).length > 1) {
+        console.log([...landmarksWithRole]);
+        console.warn(`Page contains more than one landmark with the '${role}' role and ${label} label. If two or more landmarks on a page share the same role, they must have unique labels.`);
+      }
+    }
+
+    return () => {
+      manager.removeLandmark(ref);
+    };
+  }, []);
```

Comment:
```suggestion
  useLayoutEffect(() => {
    manager.addLandmark(ref, {label, role});
    return () => {
      manager.removeLandmark(ref);
    };
  }, []);
  
  useEffect(() => {
    manager.updateLandmark({ref, label, role});
  }, [ariaLabel, ariaLabelledby]);
```

this means we can now catch if they remove the labels and we can be explicit about it in our useEffects

Reply:
Much better, thanks for the suggestion

Reply:
I think we probably want `[label, ref, role]` as deps in that useEffect though